capstone/src/main/java/com/google/sps/data/ApkFileFeatures.java

```diff
@@ -2,23 +2,27 @@
 
 import com.google.appengine.api.datastore.Entity;
 
+import java.util.ArrayList;
+import java.util.List;
+
 public class ApkFileFeatures {
 
-  // Declare variables for content in APK
-  private long dexFileSize, resFileSize, libraryFileSize, assetsFileSize, resourcesFileSize, miscFileSize, totalApkSize, filesCount;
+  // Declare variables and lists for content in APK
+  private ArrayList<Long> dexFileSize, resFileSize, libraryFileSize, assetsFileSize, resourcesFileSize, miscFileSize;
+  private Long totalApkSize, filesCount, lostSize;
 
   // Create APK class once data has been received from Datastore
   public ApkFileFeatures(Entity entity) {
-    this.dexFileSize = (long) entity.getProperty("Dex_File_Size");
```

Comment:
iirc, since you aren't passing these values in explicitly you don't need to use this.*

But consider making a second constructor that takes in all of these values explicitly (aka:

public ApkFileFeatures(long resFileSize, long assetsFileSize) {...}

This means you could test methods without explicitly creating an entity object. You could also then use that constructor in this method. This is valuable if, for example, you ever need to manually provide data (let's say by doing a batch processing or updating something).

You don't have to do this. But that is how I typically think about constructors like this.

Reply:
That's a helpful suggestion. I could use that for testing purposes while the first one could simply work with the Display Servlet